images/kindnetd/cmd/kindnetd/routes.go

```diff
@@ -24,30 +24,49 @@ import (
 	"k8s.io/klog/v2"
 )
 
-func syncRoute(nodeIP string, podCIDRs []string) error {
+func syncRoutes(nodeIP string, podCIDRs []string) error {
 	ip := net.ParseIP(nodeIP)
-
 	for _, podCIDR := range podCIDRs {
-		// parse subnet
-		dst, err := netlink.ParseIPNet(podCIDR)
-		if err != nil {
+		if err := syncRoute(podCIDR, ip); err != nil {
 			return err
 		}
+	}
+	return nil
+}
 
-		// Check if the route exists to the other node's PodCIDR
-		routeToDst := netlink.Route{Dst: dst, Gw: ip}
-		route, err := netlink.RouteListFiltered(nl.GetIPFamily(ip), &routeToDst, netlink.RT_FILTER_DST)
-		if err != nil {
-			return err
+func syncRoute(podCIDR string, gw net.IP) error {
+	// parse subnet
+	dst, err := netlink.ParseIPNet(podCIDR)
+	if err != nil {
+		return err
+	}
+
+	// List all routes to the other node's PodCIDR
+	routeToDst := netlink.Route{Dst: dst, Gw: gw}
```

Comment:
should we omit my previous mistake of using the gw?

Reply:
The `routeToDst` is still used for creating the route, but I'm ok with dropping the gw here and inline the definition below (where gw is included).

Reply:
Updated the comment for `RouteListFiltered` and `routeToDst`